### PR TITLE
Bbox crop label filtering

### PR DESCRIPTION
**Pull request by pribalta**
Add support for label filtering during bounding box selection

### CODE REVIEW

**Comment by awolant** [comment]
Maybe with this we also should update [detection pipeline example](https://github.com/NVIDIA/DALI/blob/master/docs/examples/detection_pipeline.ipynb) ?

File: dali/pipeline/operators/crop/bbox_crop.cc [hunk 1]
```diff
@@ -18,13 +18,14 @@ namespace dali {
...
 DALI_SCHEMA(RandomBBoxCrop)
     .DocStr(
-        R"code(Perform a prospective crop to an image while keeping bounding boxes consistent. Inputs must be supplied as two Tensors:
-        `Images` containing image data in NHWC format, and `BBoxes` containing bounding boxes represented as `[l,t,r,b]` or `[x,y,w,h]`.
-        Resulting prospective crop is provided as two Tensors: `Begin` containing the starting coordinates for the `crop` in `(x,y)` format,
-        and 'Size' containing the dimensions of the `crop` in `(w,h)` format. Bounding boxes are provided as a `(m*4)` Tensor,
-        where each bounding box is represented as `[l,t,r,b]` or `[x,y,w,h]`.)code")
-    .NumInput(1)
-    .NumOutput(3)
+        R"code(Perform a prospective crop to an image while keeping bounding boxes and labelsconsistent. Inputs must be supplied as
```
Comment (JanuszL) [inline dali/pipeline/operators/crop/bbox_crop.cc]: Missing space

Comment (pribalta) [reply dali/pipeline/operators/crop/bbox_crop.cc]: accepted

File: dali/pipeline/operators/crop/bbox_crop.cc [hunk 2]
```diff
@@ -91,6 +92,18 @@ void RandomBBoxCrop<CPUBackend>::WriteBoxesToOutput(
   }
 }
 
+template <>
+void RandomBBoxCrop<CPUBackend>::WriteLabelsToOutput(
+    SampleWorkspace *ws, const std::vector<int> &labels) {
+  auto *labels_out = ws->Output<CPUBackend>(3);
+  labels_out->Resize({static_cast<Index>(labels.size())});
+
+  auto *labels_out_data = labels_out->mutable_data<float>();
```
Comment (awolant) [inline dali/pipeline/operators/crop/bbox_crop.cc]: int instead of float here?

Comment (pribalta) [reply dali/pipeline/operators/crop/bbox_crop.cc]: accepted

**Comment by pribalta** [comment]
builds 39869

**Pull request merged**